teuthology/repo_utils.py

```diff
@@ -51,9 +51,9 @@ def build_git_url(project, project_owner='ceph'):
     elif project == 'ceph':
         base = config.get_ceph_git_url()
     else:
-        base = 'https://github.com/{project_owner}/{project}'
+	base = '{ceph_git_base_url}/{project_owner}/{project}'
```

Comment:
According to https://github.com/ceph/teuthology/blob/cbf9f04aba99fabd05defc976000254a396fab38/docs/siteconfig.rst?plain=1#L66

the ceph_git_base_url is already a combination of host and project owner, so this will not work as expected.